### PR TITLE
Add sqrtX96 overflow and precision analysis

### PR DESCRIPTION
**Pull request by talbaneth**

### CODE REVIEW

**Review by hexonaut — APPROVED** [review]
LGTM 👍

File: README.md [hunk 1]
```diff
@@ -31,3 +31,45 @@ Once we have the fair reserves and the prices we can compute the token price by:
...
+
+3. For the full expression not to overflow a uint160 this needs to hold:
+```
+   sqrt((p0 * 10^UNITS_1 * 2^96) / (p1 * 10^UNITS_0)) * 2^48 < 2^160
+   sqrt((p0 * 10^UNITS_1 * 2^96) / (p1 * 10^UNITS_0)) < 2^112         // 2^96 < 10^29, UNITS_1 <= 18, UNITS_0 >= 0, 2^112 > 10^33
+   sqrt((p0 * 10^18 * 10^29) / (p1 * 10^0)) < 10^33
+   sqrt((p0 * 10^18 * 10^29) / p1) < 10^33                            // ^2
+   (p0 * 10^47) / p1 < 10^66
+   (p0 / p1) < 10^19                                                  // p0 = token0_usd_price * 10^18, p1 = token1_usd_price * 10^18
+   token0_usd_price / token1_usd_price < 10^17
```
Comment (gbalabasquer) [inline README.md:74]: Am I missing something? or should this instead be:
```suggestion
   token0_usd_price / token1_usd_price < 10^19
```
?

Comment (talbaneth) [reply README.md]: right, thank you.

**Review by gbalabasquer — APPROVED** [review]
LGTM

**Comment by talbaneth** [comment]
@hexonaut I think you can merge it, I don't have write access.

**Pull request merged**